### ISSUE TITLE
Beatmap visual content should be available on beatmap info page

### ISSUE THREAD

**Issue by ubinici** [ppy/osu-web#7288]
I'm not sure if such an issue was created before, excuse me if this is a duplicate.

Currently, there's no way of obtaining the beatmap visual content without downloading the whole set and manually checking its files via the file explorer, which is a painful process we have been facing as moderators while detecting beatmaps with unwanted/inappropriate backgrounds. Even though the image previews exist on both the legacy and the current website, these usually fall short when it comes to determining their appropriateness since the legacy web provides a nano version of it and the current web provides awfully cropped ones.

This also creates a burden during content-review submissions, since reporters are expected to attach an image URL, which is already a burden in itself, and when they forget to do so the burden of resetting the voting procedure falls onto the reviewers.

In conclusion, getting beatmap visual content, especially the background, available for download or at least fetchable via API means would be a great quality of life improvement for both us, GMT & NAT, and obviously the rest of the player base.

**Comment by peppy** [comment]
I don't understand what the hard part of downloading the beatmap and unzipping it is.

Nothing we can provide will cover everything. What about backgrounds per difficulty? What about storyboard content?

**Comment by zeusminus** [comment]
It's not really hard but rather a waste of time. I'm talking about improvements that can be done for the functionality of our moderation. I'm sure that nothing can cover all the visual assets; however, even making only the backgrounds available for that matter will greatly reduce the amount of effort we usually spend.

On the other hand, storyboards very rarely occur to be inappropriate and background videos can usually be found via third-party providers, while a vast majority of content-review cases come up because of the background image or song content. The same goes for the typical BSS moderation stuff as well. Therefore, we might not even need to fetch storyboards, beatmap skins, or videos anyway. They probably won't worth the effort.

However, backgrounds per difficulty would be nice, since preview on the web page does not provide them, either. I don't know how tiresome it would be to enable those, though.

**Comment by zeusminus** [comment]
My initial intent was to have this enabled for all users, but the function will mostly be used by GMT & NAT anyway, so what you offered seems to be a decent solution.

**Comment by peppy** [comment]
I think that is something I can probably add with relative ease. Just to confirm, you would want to see all backgrounds across all difficulties?

**Comment by Joehuu** [comment]
Related to https://github.com/ppy/osu-web/issues/2687.

**Comment by zeusminus** [comment]
> Just to confirm, you would want to see all backgrounds across all difficulties?

Preferably, yes. That would most certainly ease things.

**Comment by peppy** [comment]
I'll add this to my bucket list of tasks. Assuming it's not super-high-priority but more something nice to have?

**Comment by zeusminus** [comment]
Indeed, it's more of a quality-of-life type improvement. Perhaps can be implemented alongside other tools internally discussed and brought to your attention.

**Comment by peppy** [comment]
Touching back on this, I think we could expose the cover admin page (ie. <https://osu.ppy.sh/admin/beatmapsets/1670776/covers>) to the content moderation teams using the existing admin popup button similar to wiki pages. In discussion with the team this would be more useful than nothing at all, and is likely the most we can offer for the time being anyway.

@nanaya @notbakaneko maybe a quick one to add?

**Comment by Hiviexd** [comment]
any progress on this? seems like it slipped through the cracks ;-;

**Comment by nanaya** [comment]
I don't think backgrounds apart of the first one are accessible without extracting the map file? I suppose anyone checking the background would want to make sure all of them are appropriate? And I don't think osu-web can tell if there are multiple backgrounds without downloading the file first either

**Comment by peppy** [comment]
Yeah, that's fine. Just wanted to expose the minimum we can with little effort as per https://github.com/ppy/osu-web/issues/7288#issuecomment-1050500060.

TL;DR making the cover admin page available to moderators via the hidden admin button thing.

**Comment by nanaya** [comment]
would a link to full size cover image be enough? Or does it need to be the cover admin page? (including regen and delete buttons)

**Comment by peppy** [comment]
Feels a bit raw to link to the full cover rather than the regen page (although the delete button should probably not be there under any circumstance – the deletion flow is handled elsewhere these days).
